playground/Testcases.tsx

```diff
@@ -0,0 +1,114 @@
+import { useRef, useEffect, Fragment } from "react";
+
+import { FLOWCHART_DIAGRAM_TESTCASES } from "./testcases/flowchart";
+import { SEQUENCE_DIAGRAM_TESTCASES } from "./testcases/sequence.ts";
+import { CLASS_DIAGRAM_TESTCASES } from "./testcases/class.ts";
+import { UNSUPPORTED_DIAGRAM_TESTCASES } from "./testcases/unsupported.ts";
+
+import type { UpdateMermaidDefinition } from "./index.tsx";
+import { MermaidDiagram } from "./MermaidDiagram.tsx";
+
+interface TestCaseProps {
+  name: string;
+  testcases: { name: string; definition: string }[];
+  onChange: (definition: string, activeTestcaseIndex: number) => void;
+  error: string | null;
+  activeTestcase?: number;
+}
+
+const Testcase = ({
+  name,
+  testcases,
+  onChange,
+  error,
+  activeTestcase,
+}: TestCaseProps) => {
+  const baseId = name.toLowerCase();
+
+  useEffect(() => {
+    if (activeTestcase !== undefined) {
+      const { definition } = testcases[activeTestcase];
+
+      onChange(definition, activeTestcase);
+    }
+  }, [testcases]);
+
+  return (
+    <>
+      <h2>
+        {name} {"Diagrams"}
+      </h2>
+      <details>
+        <summary>
+          {name} {"Examples"}
+        </summary>
+        <div id={`${baseId}-container`} className="testcase-container">
+          {testcases.map(({ name, definition }, index) => {
+            const id = `${baseId}-${index}`;
+            return (
+              <Fragment key={id}>
+                <h2 style={{ marginTop: "50px", color: "#f06595" }}>{name}</h2>
+                <pre>{definition}</pre>
+                <button
+                  onClick={() => {
+                    onChange(definition, index);
+                  }}
+                >
+                  {"Render to Excalidraw"}
+                </button>
+
+                <MermaidDiagram definition={definition} id={id} />
+
+                {error && activeTestcase === index && (
```

Comment:
This would mean it shows error only for active testcase lets make it render error irrespective of the condition ?

Currently If I edit a testcase file such with errored syntax, the text gets updated however mermaid diagram doesn't rerender.

Reply:
@ad1992 I was thinking about this, and the idea of ​​just showing for the active test case was to not have to store the individual errors for each testcase.

Today we don't have the reference so the last error would be overwritten, so that error structure in the App file would become a kind of multidimensional array:
```js
[
  [
    0 // index testcases files, 
    0 // index testcase inside testscases file
  ]
]
```

Reply:
It got a lot better hahaha
done in [2e09454](https://github.com/excalidraw/mermaid-to-excalidraw/pull/52/commits/2e094543b605521b5ce94085302dd7879867c59a)

Reply:
I kept it in the custom test case so we could track the parseMermaid, not just the diagram error.